test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
+    std::vector<Source> sources;
+    Group append_group = block.createGroup("ag", "test");
+    Group set_group = block.createGroup("sg", "test");
+
+    for (auto it = names.begin(); it != names.end(); it++) {
+        sources.push_back(block.createSource(*it, "channel"));
+    }
+
+    // append each source in order
+    for (auto src : block.sources()) {
```

Comment:
`block.sources()` returns a `vector<Source>`, so src is of type `Source`, which is ok (we are allocating a Source on the stack and copying a `shared_ptr<ISource>` (via the copy constructor(s)); slightly better, `const auto &src`, now unsurprisingly `src` is of type `const Source &` and we are basically just passing a pointer around. Totally ok to not do in a test, also allowed ;)